test/extended/prometheus/prometheus_ocm_workqueue.go

```diff
@@ -0,0 +1,85 @@
+package prometheus
+
+import (
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/ibmcloud"
+)
+
+var _ = g.Describe("[Feature:Prometheus][Feature:Builds] Prometheus", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithoutNamespace("prometheus")
+
+		url, bearerToken string
+	)
+	g.BeforeEach(func() {
+		var ok bool
+		url, bearerToken, ok = locatePrometheus(oc)
+		if !ok {
+			e2e.Skipf("Prometheus could not be located on this cluster, skipping prometheus test")
+		}
+	})
+
+	g.AfterEach(func() {
+		if g.CurrentGinkgoTestDescription().Failed {
+			exutil.DumpPodStatesInNamespace("openshift-monitoring", oc)
+			exutil.DumpPodLogsStartingWithInNamespace("prometheus-k8s", "openshift-monitoring", oc)
+		}
+	})
+
+	g.Describe("when installed on the cluster", func() {
+		g.It("should start and expose a secured proxy and verify OCM workqueue metrics", func() {
+
+			// https://issues.redhat.com/browse/CO-895
+			if e2e.TestContext.Provider == ibmcloud.ProviderName {
+				e2e.Skipf("Prometheus in IBM ROKS clusters does not collect metrics from the OpenShift controller manager because it lives outside of the cluster. " +
+					"The openshift_build_total metric expected by this test is reported by the OpenShift controller manager. " +
+					"Remove this skip when https://issues.redhat.com/browse/CO-895 is implemented.")
+			}
+
+			oc.SetupProject()
+			ns := oc.Namespace()
+
+			execPod := exutil.CreateCentosExecPodOrFail(oc.AdminKubeClient(), ns, "execpod", nil)
+			defer func() { oc.AdminKubeClient().CoreV1().Pods(ns).Delete(execPod.Name, metav1.NewDeleteOptions(1)) }()
+
+			g.By("verifying the oauth-proxy reports a 403 on the root URL")
+			// allow for some retry, a la prometheus.go and its initial hitting of the metrics endpoint after
+			// instantiating prometheus tempalte
+			var err error
+			for i := 0; i < maxPrometheusQueryAttempts; i++ {
+				err = expectURLStatusCodeExec(ns, execPod.Name, url, 403)
+				if err == nil {
+					break
+				}
+				time.Sleep(prometheusQueryRetrySleep)
+			}
+			o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
Is the above step needed to ensure that the `oauth-proxy` is up and the service account token will authenticate?

Reply:
the above is the endpoint/svc is up, the code below is related to the SA token

it mimics what is done in prometheus_builds.go 

prometheus.go has something similar, but without the retry on the 403 ... as I recall my testing back in 3.10 for builds seemed to hit timing windows the vanilla prometheus did not ... the diff goes back that far ... but it has been a while